### PR TITLE
bm25 datasets notebook

### PR DESCRIPTION
**Pull request by acatav**

### CODE REVIEW

**Comment by gdj0nes** [comment]
> made changes to bm25-quora notebook, haven't had a chance to go through vector-gen notebook yet but on quick skim through, I saw we're importing `pinecone_text` but I didn't see the `pip install`? I can look through it properly at 13:00 ET

Resolved

File: pinecone/sparse/bm25/bm25-quora.ipynb [hunk 1]
```diff
@@ -0,0 +1,2451 @@
+{
```
Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-quora.ipynb:1]: <p>"It is (a) simple but effective and only requires knowing the number of documents in a corpus and the frequency of terms across documents."</p><p> </p><p>"In the following guide[,] ..."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-quora.ipynb:1]: <blockquote>We'll install the required libraries: the <code>pinecone-client</code> for interacting with Pinecone, the <code>pinecone-datasets</code> library that we will use for fast processing of the Quora dataset, and <code>numpy</code>.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-quora.ipynb:1]: <p>We'll load the popular Quora dataset with precomputed embeddings. Both dense and sparse embeddings have been precomputed using the following models: </p><ul><li>Dense:&nbsp;<a href="https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2" target="_blank">sentence-transformers/all-MiniLM-L6-v2</a></li><li><br></li><li>Sparse: BM25</li></ul><blockquote><br></blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-quora.ipynb:1]: <blockquote>As you can see, this data is already loaded with the sparse and dense representations of each document. To learn about the generation process of this values, see <a href="https://github.com/pinecone-io/examples/blob/master/pinecone/sparse/bm25/bm25-vector-generation.ipynb" target="_blank">this walkthrough</a>.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-quora.ipynb:1]: <blockquote>For this, we need a <a href="https://app.pinecone.io/" target="_blank">free API key</a>. We initialize the connection like so: </blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-quora.ipynb:1]: <blockquote>"And we connect to the index like so:"</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-quora.ipynb:1]: <blockquote>Now let's upsert vectors to the index. We are using async upload with batching. For more information on performance boosting, see the Pinecone documentation for <a href="https://docs.pinecone.io/docs/performance-tuning" target="_blank">Performance Tuning</a>.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-quora.ipynb:1]: <blockquote>"Here we define a function that merges the query results with the actual texts of the documents and shows them as a dataframe."</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-quora.ipynb:1]: <blockquote>" In the following code, we explore the impact of various alpha values using a sample query."</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

File: pinecone/sparse/bm25/bm25-vector-generation.ipynb [hunk 2]
```diff
@@ -0,0 +1,6251 @@
+{
```
Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-vector-generation.ipynb:1]: <p>"It is (a) simple but effective..."</p><p>"In the following guide, ..."</p><p>"sparse-dense index" -&gt; "sparse-dense vectors" (There's no specific index for sparse-dense)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-vector-generation.ipynb:1]: <p>Add period to end of sentence</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-vector-generation.ipynb:1]: <p>Add period to end of sentence</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-vector-generation.ipynb:1]: <p>Sp: 'Organize/Organise' </p><p>Align to -&gt; align with</p><p>Period</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-vector-generation.ipynb:1]: <p>Capitalization: [F]or more details...</p><p>Comma after "on that[,]..."</p><p>Two words: "checkout" -&gt; "check out"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

File: pinecone/sparse/splade/splade_quora.ipynb [hunk 3]
```diff
@@ -0,0 +1,2438 @@
+{
```
Comment (byronnlandry) [inline pinecone/sparse/splade/splade_quora.ipynb:1]: <p>Load the popular Quora dataset with embeddings precomputed using the following models:</p><ul><li>Dense:&nbsp;<a href="https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2" target="_blank">sentence-transformers/all-MiniLM-L6-v2</a></li><li>Sparse: SPLADE</li></ul><blockquote><br></blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_quora.ipynb:1]: <blockquote>We first need to initialize our connection to Pinecone to create our vector index. For this, we need a free <a href="https://github.com/pinecone-io/examples/raw/d9d51bddf60c0f6588251c5fccd6165b4d300613/pinecone/sparse/splade/%22https://app.pinecone.io/%22" target="_blank">API key</a>. We initialize the connection like so: </blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_quora.ipynb:1]: <p>Now let's upsert vectors to the index. We are using async upload with batching. For more imformation on performance boosting, visit the Pinecone documentation on <a href="https://docs.pinecone.io/docs/performance-tuning" target="_blank">performance tuning</a>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_quora.ipynb:1]: <blockquote>First let's look at our dataset queries.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_quora.ipynb:1]: <blockquote>Here we define a function that merges the query results with the actual texts of the documents and shows them as a dataframe.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

File: pinecone/sparse/splade/splade_vector_generation.ipynb [hunk 4]
```diff
@@ -0,0 +1,8784 @@
+{
```
Comment (byronnlandry) [inline pinecone/sparse/splade/splade_vector_generation.ipynb:1]: <p>"Unlike dense embeddings[,] which can be difficult to interpret[,] sparse embeddings map to tokens for easier interpretability."</p><p> </p><p>"sparse-dense index" -&gt; "sparse-dense vectors" (No special index for sparse-dense)</p><p> </p><p>Omit: ["See the companion guide to learn to skip embedding generation"] (redundant)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_vector_generation.ipynb:1]: <p>Add period to end of sentence</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_vector_generation.ipynb:1]: <p>"In the following example[,] we will use(.) [the] SPLADE Model: <a href="https://huggingface.co/naver/splade-cocondenser-ensembledistil" target="_blank">naver/splade-cocondenser-ensembledistil</a> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_vector_generation.ipynb:1]: <p>Add period to end of sentence</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_vector_generation.ipynb:1]: <p>"Orgnise"-&gt;"Organize|Organise"</p><p>"align to" -&gt; "align with the"</p><p>Add period to end of sentence</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_vector_generation.ipynb:1]: <p>"[F]or more details on that[,] check[ ]out this notebook."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

File: pinecone/sparse/bm25/bm25-quora.ipynb (hunk 1 above)
Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-quora.ipynb:1]: <p>"It is (a) simple but effective and only requires knowing the number of documents in a corpus and the frequency of terms across documents."</p><p> </p><p>"In the following guide[,] ..."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-quora.ipynb:1]: <blockquote>We'll install the required libraries: the <code>pinecone-client</code> for interacting with Pinecone, the <code>pinecone-datasets</code> library that we will use for fast processing of the Quora dataset, and <code>numpy</code>.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-quora.ipynb:1]: <p>We'll load the popular Quora dataset with precomputed embeddings. Both dense and sparse embeddings have been precomputed using the following models: </p><ul><li>Dense:&nbsp;<a href="https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2" target="_blank">sentence-transformers/all-MiniLM-L6-v2</a></li><li><br></li><li>Sparse: BM25</li></ul><blockquote><br></blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-quora.ipynb:1]: <blockquote>As you can see, this data is already loaded with the sparse and dense representations of each document. To learn about the generation process of this values, see <a href="https://github.com/pinecone-io/examples/blob/master/pinecone/sparse/bm25/bm25-vector-generation.ipynb" target="_blank">this walkthrough</a>.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-quora.ipynb:1]: <blockquote>For this, we need a <a href="https://app.pinecone.io/" target="_blank">free API key</a>. We initialize the connection like so: </blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-quora.ipynb:1]: <blockquote>"And we connect to the index like so:"</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-quora.ipynb:1]: <blockquote>Now let's upsert vectors to the index. We are using async upload with batching. For more information on performance boosting, see the Pinecone documentation for <a href="https://docs.pinecone.io/docs/performance-tuning" target="_blank">Performance Tuning</a>.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-quora.ipynb:1]: <blockquote>"Here we define a function that merges the query results with the actual texts of the documents and shows them as a dataframe."</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-quora.ipynb:1]: <blockquote>" In the following code, we explore the impact of various alpha values using a sample query."</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

File: pinecone/sparse/bm25/bm25-vector-generation.ipynb (hunk 2 above)
Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-vector-generation.ipynb:1]: <p>"It is (a) simple but effective..."</p><p>"In the following guide, ..."</p><p>"sparse-dense index" -&gt; "sparse-dense vectors" (There's no specific index for sparse-dense)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-vector-generation.ipynb:1]: <p>Add period to end of sentence</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-vector-generation.ipynb:1]: <p>Add period to end of sentence</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-vector-generation.ipynb:1]: <p>Period</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-vector-generation.ipynb:1]: <p>Sp: 'Organize/Organise' </p><p>Align to -&gt; align with</p><p>Period</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-vector-generation.ipynb:1]: <p>Capitalization: [F]or more details...</p><p>Comma after "on that[,]..."</p><p>Two words: "checkout" -&gt; "check out"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

File: pinecone/sparse/splade/splade_quora.ipynb (hunk 3 above)
Comment (byronnlandry) [inline pinecone/sparse/splade/splade_quora.ipynb:1]: <blockquote>Unlike dense embeddings, which can be difficult to interpret, sparse embeddings map to tokens for easier interpretability.</blockquote><blockquote>See the <a href="https://github.com/pinecone-io/examples/blob/master/pinecone/sparse/splade/splade_vector_generation.ipynb" target="_blank">companion guide</a> to learn how to generate embeddings. &lt;--period</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_quora.ipynb:1]: <p>Load the popular Quora dataset with embeddings precomputed using the following models:</p><ul><li>Dense:&nbsp;<a href="https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2" target="_blank">sentence-transformers/all-MiniLM-L6-v2</a></li><li>Sparse: SPLADE</li></ul><blockquote><br></blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_quora.ipynb:1]: <p>This should link to the embedding notebook</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_quora.ipynb:1]: <blockquote>We first need to initialize our connection to Pinecone to create our vector index. For this, we need a free <a href="https://github.com/pinecone-io/examples/raw/d9d51bddf60c0f6588251c5fccd6165b4d300613/pinecone/sparse/splade/%22https://app.pinecone.io/%22" target="_blank">API key</a>. We initialize the connection like so: </blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_quora.ipynb:1]: <p>Now let's upsert vectors to the index. We are using async upload with batching. For more imformation on performance boosting, visit the Pinecone documentation on <a href="https://docs.pinecone.io/docs/performance-tuning" target="_blank">performance tuning</a>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_quora.ipynb:1]: <blockquote>First let's look at our dataset queries.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_quora.ipynb:1]: <blockquote>Here we define a function that merges the query results with the actual texts of the documents and shows them as a dataframe.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

File: pinecone/sparse/splade/splade_vector_generation.ipynb (hunk 4 above)
Comment (byronnlandry) [inline pinecone/sparse/splade/splade_vector_generation.ipynb:1]: <p>"Unlike dense embeddings[,] which can be difficult to interpret[,] sparse embeddings map to tokens for easier interpretability."</p><p> </p><p>"sparse-dense index" -&gt; "sparse-dense vectors" (No special index for sparse-dense)</p><p> </p><p>Omit: ["See the companion guide to learn to skip embedding generation"] (redundant)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_vector_generation.ipynb:1]: <p>Add period to end of sentence</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_vector_generation.ipynb:1]: <p>"In the following example[,] we will use(.) [the] SPLADE Model: <a href="https://huggingface.co/naver/splade-cocondenser-ensembledistil" target="_blank">naver/splade-cocondenser-ensembledistil</a> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_vector_generation.ipynb:1]: <p>Add period to end of sentence</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_vector_generation.ipynb:1]: <p>"Orgnise"-&gt;"Organize|Organise"</p><p>"align to" -&gt; "align with the"</p><p>Add period to end of sentence</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade_vector_generation.ipynb:1]: <p>"[F]or more details on that[,] check[ ]out this notebook."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

**Review by jamescalam — APPROVED** [review]
lgtm

@byronnlandry I think I covered all of the language suggestions from you, but probably worth one more check from your side if anything is missing

File: pinecone/sparse/bm25/bm25-quora.ipynb (hunk 1 above)
Comment (byronnlandry) [reply pinecone/sparse/bm25/bm25-quora.ipynb:1]: <p>Still missing a comma after "following guide"</p>

File: pinecone/sparse/bm25/bm25-vector-generation.ipynb [hunk 5]
```diff
@@ -0,0 +1,6294 @@
...
```
Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-vector-generation.ipynb:1]: <p>"Spacy | spacy" -&gt; "spaCy"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-vector-generation.ipynb:1]: <p>spacy -&gt; spaCy</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-vector-generation.ipynb:1]: <p>To do this[,]...</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (byronnlandry) [inline pinecone/sparse/bm25/bm25-vector-generation.ipynb:1]: <p>...vector database {. | !}</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

File: pinecone/sparse/splade/splade-quora.ipynb [hunk 6]
```diff
@@ -0,0 +1,2439 @@
+{
```
Comment (byronnlandry) [inline pinecone/sparse/splade/splade-quora.ipynb:1]: <blockquote>Unlike dense embeddings, which can be difficult to interpret, sparse embeddings map to tokens for easier interpretability.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

File: pinecone/sparse/splade/splade-vector-generation.ipynb [hunk 7]
```diff
@@ -0,0 +1,8782 @@
+{
```
Comment (byronnlandry) [inline pinecone/sparse/splade/splade-vector-generation.ipynb:1]: <p>...following example[,]</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (byronnlandry) [inline pinecone/sparse/splade/splade-vector-generation.ipynb:1]: <p>...vector database { . | ! }</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/138/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

**Pull request merged**